### ISSUE TITLE
CMSIS RTOS Validator: Silence Warnings

### ISSUE THREAD

**Issue by ilg-ul** [ARM-software/CMSIS#21]
Compiling the files in the Source folder triggered 153 warnings.

Suggested changes:

- fix the code to avoid warnings.

Workaround:

- to silence the warnings, the following GCC options can be use when compiling the files in the Source folder:

```
-Wno-aggregate-return -Wno-conversion -Wno-unused-parameter \
-Wno-missing-prototypes -Wno-missing-declarations -Wno-sign-compare \
-Wno-maybe-uninitialized -Wno-unused-function -Wno-format -Wno-padded
```

Rationale:

- the CMSIS++ code builds with all available warnings enabled.

**Comment by brondani** [comment]
It is unclear how to reproduce that as I only get a single warning. However this might be because I'm using new CMSIS-Core files or used a different GCC.

**Comment by ilg-ul** [comment]
by default my projects are configured with almost all available warnings, as seen in the attached screenshots.
![screen shot 2016-04-15 at 21 45 45](https://cloud.githubusercontent.com/assets/362732/14571774/ba626972-0353-11e6-9987-169da90ef75b.png)
![screen shot 2016-04-15 at 21 45 54](https://cloud.githubusercontent.com/assets/362732/14571775/ba892df0-0353-11e6-82da-5cc3b2537ca8.png)
![screen shot 2016-04-15 at 21 46 05](https://cloud.githubusercontent.com/assets/362732/14571776/ba8d3c42-0353-11e6-919a-e739022e6a82.png)

**Comment by ilg-ul** [comment]
could you confirm being able to reproduce the issue, using the attached images?

**Comment by brondani** [comment]
Yes, about 150 warnings are triggered when adding the warning options from your attached images, namely:
-Wmissing-prototypes
-Wstrict-prototypes 
-Wbad-function-cast 
-Wunused 
-Wuninitialized 
-Wall 
-Wextra 
-Wmissing-declarations 
-Wconversion 
-Wpointer-arith 
-Wpadded 
-Wshadow 
-Wlogical-op 
-Waggregate-return 
-Wfloat-equal

Anyway if possible please attach your list of warnings, thanks.

**Comment by ilg-ul** [comment]
do you really need the actual compiler output? as for the list of distinct warnings, the list of 'counter-measures' I used to avoid your warnings is in the initial ticket text, you can deduce the warnings from there.

**Comment by ilg-ul** [comment]
btw, could you make an internal memo so that all code issued by ARM/Keil is also checked in this configuration? (gcc & all those warnings active). 

for running tests, I also suggest using `-Werror`, to break the build on any warning.

**Comment by brondani** [comment]
Your compiler output is not necessary, just in case you have it nearby.
And yes I can proceed with an internal memo.
Thank you again for your suggestions.
